resources/task.rb

```diff
@@ -0,0 +1,149 @@
+  end
+end
+
+action :create do
+  init
+
+  converge_if_changed do
+    nexus3_api "upsert_task #{new_resource.task_name}" do
+      script_name 'upsert_task'
+      args name: new_resource.task_name,
+           source: new_resource.task_source,
+           crontab: new_resource.task_crontab
+
+      action %i(create run)
+      endpoint new_resource.api_endpoint
+      username new_resource.api_username
+      password new_resource.api_password
+
+      content <<-EOS
+// Freely adapted from
+// https://github.com/savoirfairelinux/ansible-nexus3-oss/blob/master/files/groovy/create_task.groovy
+import org.sonatype.nexus.scheduling.TaskConfiguration;
+import org.sonatype.nexus.scheduling.TaskInfo;
+import org.sonatype.nexus.scheduling.TaskScheduler;
+import org.sonatype.nexus.scheduling.schedule.Schedule;
+
+import groovy.json.JsonSlurper;
+
+def params = new JsonSlurper().parseText(args);
+
+TaskScheduler taskScheduler = container.lookup(TaskScheduler.class.getName());
+TaskInfo existingTask = taskScheduler.listsTasks().find { TaskInfo taskInfo ->
+   taskInfo.getName() == params.name;
+}
+if (existingTask && !existingTask.remove()) {
+   throw new RuntimeException("Could not remove currently running task: " + params.name);
+}
+
+TaskConfiguration taskConfiguration = taskScheduler.createTaskConfigurationInstance('script');
+taskConfiguration.setName(params.name);
+taskConfiguration.setString('source', params.source);
+Schedule schedule = taskScheduler.scheduleFactory.cron(new Date(), params.crontab);
+
+taskScheduler.scheduleTask(taskConfiguration, schedule);
+      EOS
+    end
+  end
+end
+
+action :delete do
+  init
+
+  converge_if_changed do
+    nexus3_api "delete_task #{new_resource.task_name}" do
+      action %i(create run)
+      script_name 'delete_task'
+      args new_resource.task_name
+
+      content <<-EOS
+import groovy.json.JsonSlurper;
+
+import org.sonatype.nexus.scheduling.TaskConfiguration;
+import org.sonatype.nexus.scheduling.TaskInfo;
+import org.sonatype.nexus.scheduling.TaskScheduler;
+
+TaskScheduler taskScheduler = container.lookup(TaskScheduler.class.getName());
+TaskInfo existingTask = taskScheduler.listsTasks().find { TaskInfo taskInfo ->
+    taskInfo.getName() == args;
+}
+
+if (existingTask && !existingTask.remove()) {
+   throw new RuntimeException("Could not remove currently running task: " + args);
+}
+    EOS
+
+      endpoint new_resource.api_endpoint
+      username new_resource.api_username
+      password new_resource.api_password
+
+      not_if { current_resource.nil? }
+    end
+  end
+end
+
+action_class do
+  def init
+    chef_gem 'httpclient' do
+      compile_time true
+    end
+
+    nexus3_api "get_task #{new_resource.task_name}" do
+      action :create
+      script_name 'get_task'
+      args new_resource.task_name
+      endpoint new_resource.api_endpoint
+      username new_resource.api_username
+      password new_resource.api_password
+
+      content <<-EOS
```

Comment:
Move the content to the file directory and use `::Nexus3::Scripts.groovy_content('get_task', node)`

Reply:
done